ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -106,7 +107,7 @@
     this.diskIOScheduler = diskIOScheduler;
     this.time = time;
     scanSizeInBytes = Math.min(config.storeCleanupOperationsBytesPerSec * 10, 1024 * 1024);
-    messageRetentionSeconds = config.storeDeletedMessageRetentionDays * Time.SecsPerDay;
+    messageRetentionSeconds = Math.toIntExact(TimeUnit.DAYS.toSeconds(Integer.toUnsignedLong(config.storeDeletedMessageRetentionDays)));
```

Comment:
the TimeUnit.DAYS.toSeconds https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html#toSeconds(long) accepts and returns long data type. Isn't that an issue ?